### ISSUE TITLE
BUG:  setting the default grid properties via template is broken in new versions of the obsidian plugin

### ISSUE THREAD

**Issue by recspert** [zsviczian/obsidian-excalidraw-plugin#2032]
### Have you searched for existing issues (including closed ones)?

- [X] Yes, I have searched the existing issues.

### Does this bug persist in a new vault with only Excalidraw installed?

- [X] Yes, I have verified the issue persists.

### Your environment

SYSTEM INFO:
	Obsidian version: v1.6.7
	Installer version: v1.6.7
	Operating system: Windows 11 Home 10.0.22631
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Base theme: dark
	Community theme: Border v1.9.8
	Snippets enabled: 0
	Restricted mode: off
	Plugins installed: 26
	Plugins enabled: 19
		1: Dataview v0.5.67
		2: Calendar v1.5.10
		3: Annotator v0.2.11
		4: Tag Wrangler v0.6.1
		5: Copy as HTML v1.1.3
		6: Text Format v3.1.0
		7: Tags Overview v1.0.4
		8: Image Uploader v0.3.2
		9: Enhancing Export v1.10.7
		10: QuickShare v1.3.1
		11: PDF++ v0.40.9
		12: Nested tags graph v1.0.2
		13: Media Notes v1.3.0
		14: Advanced Tables v0.22.0
		15: Various Complements v10.0.3
		16: Pane Relief v0.5.5
		17: Style Settings v1.0.9
		18: Excalidraw v2.5.0
		19: Importer v1.6.5

### Describe the bug

Changing the default grid size is tricky in the Obsidian plugin. I used to define grid properties such as size and major grid frequency via a template file `template.excalidraw.md`, which contained the following lines (under the `Drawing` section):

# Drawing (old)
```json
{
	"type": "excalidraw",
	"version": 2,
	"source": "https://github.com/zsviczian/obsidian-excalidraw-plugin/releases/tag/1.9.26",
	"elements": [],
	"appState": {
		...
		"gridSize": 9,
		"gridColor": {
			"Bold": "#C9C9C9FF",
			"Regular": "#EDEDEDFF",
			"MajorGridFrequency": 2
		},
		...
}
```

After I installed the fresh version of the plugin, the template file has been overwritten and apparently all my previous settings are gone. Moreover, the template file now contains some hash-values instead of human-readable json, so it's unclear how to set the new defaults for the grid properties. Here's how it looks like now:

# Drawing (new)
```
N4KAkARALgngDgUwgLgAQQQDwMYEMA2AlgCYBOuA7hADTgQBuCpAzoQPYB2KqATLZMzYBXUtiRoIACyhQ4zZAHoFAc0JRJQgEYA6bGwC2CgF7N6hbEcK4OCtptbErHALRY8RMpWdx8Q1TdIEfARcZgRmBShcZQUebQBWbQAGGjoghH0EDihmbgBtAF1+CFw4OABlKKhxVFAwSHUMmohiXFIAa1T6hkIECgAhXGx25VJhDmIAYTZ8NlJuCABiADNV

ta7IbBFA7IBJav1KsfaEadn5iUWARgQbm42ILdIdqH2MweHR8amZuYWocgcZhwNpZKAPJ4vN76ABihHw+EqMGCC0kQw0gQh2zB0KObBOAHUSOpuEltAAWfibbF7A54k5IlESQQeLHPHEHABKwmUkg44VyaCuVMeNNeBwA8iDsGoYNwrkkkiLIRyMjDOFAYbh9PC5ULlWLoersuVCEYajwleBqezaRkACpYKAAQSIyi4EmCy3BBtt4oylTaLuebAo
 ...
 ```

### Steps to reproduce

1. Open a new drawing.
2. Observe the grid size properties are different from what was previously set via the json configuration in the template file.

### Expected behavior

The grid size properties defined via a template file must be respected. Migration to newer versions of the plugin must copy the custom default settings in the template file or warn the user about possible changes and provide instruction on how to preserve the settings. 

Unfortunately, it won't help address the current issue as I have no idea on how to get my custom grid settings back into the template so that I do not have to change grid size properties manually each time I open a new drawing.

**Comment by recspert** [comment]
Hi, thanks for the quick response and for the great app! The trick with decompressing the json did the job.

Just to add a few points, which may save some time for others reading this issue. 
The `Set Grid Size` script doesn't change the default values, it requires setting the grid size each time one opens a new drawing. My question was about _changing the defaults_, so that the desired grid size is set only _once and forever_ for all new drawings. The script doesn't help with that, unfortunately.

The `Canvas & Shape Properties` menu also does not help resolving the issue. Editing the grid via this menu on a template file only changes the grid size temporarily (i.e., while the drawing is open). Once the template file is closed, the settings get restored back to original values. After playing around for a while, I was able to make it store the settings permanently by explicitly hitting `Ctrl+S` to save the file before closing it. This step was not obvious to me as markdown files do not normally require explicitly saving changes to them. Moreover, the template file doesn't indicate in any way that there are some user-set settings that won't be saved if the file is closed.

So, the behavior of editing the grid via `Canvas & Shape Properties` on a template file is confusing. I would suggest considering the following improvements:
1) (preferred) change the behavior so that editing via `Canvas & Shape Properties` takes permanent effect and overrides the defaults of the template immediately without requiring a user to explicitly save the template file;
2) warn a user about unsaved changes (or at least change the color of the save icon to red like when a user draws something).

I guess, this would help avoid confusion.

Thanks again for the work!

**Comment by zsviczian** [comment]
Grid settings are saved. However, if you make no other changes to the drawing environment changes such as grid setting or selecting the lock tool and some others aren't registered as changes, thus they are not automatically saved. If the only changes you make to a template file are environmental changes then you need to press CTRL+S to force save the drawing. If you make other changes as well, then grid changes will be saved.

**Comment by recspert** [comment]
Yes, this is exactly what I'm pointing out. The new way the app behaves is convenient _only once you know_ that you _have to save the template_ after changing environmental settings. However, unless you know it (and I would argue that this is the default state of most new users), the behavior is confusing.

That's exactly what happened to me. I was trying to change grid settings on a template file via the `Canvas & Shape Properties`. I assumed that after I set new values for the grid settings they are saved automatically, because the save icon in the top right corner is not red (unlike the cases when you draw something). However, after closing the file the newly set settings were gone, which confused me and led to opening this issue.

That's why I suggested to consider 2 modifications to the behavior whichever is more plausible/easy to do. However, I understand that it can be time consuming. Maybe it can go at least to a low-priority list of possible future enhancements. That would be great. I believe, it would reduce possible confusion for many users.

Probably, the third viable option is to describe somewhere in the documentation what you described in the previous answer. Maybe in the dedicated section related to templates, because this is where the risk of confusion seems to be the highest.
